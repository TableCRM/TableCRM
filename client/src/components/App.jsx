import React from 'react';
import { Link, Switch, Route } from 'react-router-dom';
import Dashboard from './Dashboard.jsx';
import Nav from './Nav.jsx';
import Leads from '../container/Leads.jsx';
import Opportunities from '../container/Opportunities.jsx';
import Contacts from '../container/Contacts.jsx';
import Accounts from '../container/Accounts.jsx';
import People from './People.jsx';
import Companies from './Companies.jsx';

class App extends React.Component {
  render() {
    return (
      <div>
        <Nav />
        <Switch>
          <Route exact path="/dashboard" component={Dashboard} />
          <Route exact path="/leads" component={Leads} />
          <Route exact path="/opportunities" component={Opportunities} />
          <Route exact path="/contacts" component={Contacts} />
          <Route exact path="/accounts" component={Accounts} />
        </Switch>
      </div>
    );
  }
}

export default App;
