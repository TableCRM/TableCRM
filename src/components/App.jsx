import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Dashboard from './Dashboard.jsx';
import Leads from '../container/Leads.jsx';
import Opportunities from '../container/Opportunities.jsx';
import People from './People.jsx';
import Companies from './Companies.jsx';

// class App extends React.Component {
//   render() {
//     return (
//       <div>
//         <div>Header</div>
//         <Switch>
//           <Route exact path="/dashboard" component={Dashboard} />
//           <Route exact path="/people" component={People} />
//           <Route exact path="/companies" component={Companies} />
//           <Route exact path="/leads" component={Leads} />
//           <Route exact path="/opportunities" component={Opportunities} />
//         </Switch>
//       </div>
//     );
//   }
// }
//
// export default App;
