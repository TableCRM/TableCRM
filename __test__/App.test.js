// This is a Jest file for testing the App Component

import React from 'react';
import { shallow } from 'enzyme';
import App from '../client/src/components/App';

const app = shallow(<App />);

it('renders correctly', () => {
  expect(app).toMatchSnapshot();
});
