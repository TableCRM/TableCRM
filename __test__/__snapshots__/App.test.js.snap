// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          Header
        </div>,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/dashboard"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/people"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/companies"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/leads"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/opportunities"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/contacts"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/accounts"
          />
        </Switch>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Header",
        },
        "ref": null,
        "rendered": "Header",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/dashboard"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/people"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/companies"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/leads"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/opportunities"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/contacts"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/accounts"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/dashboard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/people",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/companies",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/leads",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/opportunities",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/contacts",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/accounts",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Header
          </div>,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/dashboard"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/people"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/companies"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/leads"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/opportunities"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/contacts"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/accounts"
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Header",
          },
          "ref": null,
          "rendered": "Header",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/dashboard"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/people"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/companies"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/leads"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/opportunities"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/contacts"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/accounts"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/dashboard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/people",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/companies",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/leads",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/opportunities",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/contacts",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/accounts",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
